packages/3dom/src/context-iframe.ts

```diff
@@ -0,0 +1,79 @@
+import {ThreeDOMCapability} from './api.js';
+import {ThreeDOMExecutionContext} from './context.js';
+import {ThreeDOMMessageType} from './protocol.js';
+
+// This semi fake worker is used to capture the post message of the handshake
+// and delay it until the port is given.
+class IFrameWorker extends Worker {
+  protected handshakeMessage: any;
+  protected port2: MessagePort|null = null;
+
+  constructor(url: string) {
+    super(url);
+  }
+
+  // Override
+  postMessage(message: any, transfer: Array<Transferable>): void;
+  postMessage(message: any, options?: PostMessageOptions|undefined): void;
+  postMessage(message: any, secondArg: any): void {
+    if (message.type == ThreeDOMMessageType.HANDSHAKE) {
+      this.handshakeMessage = message;
+      this.postHandshakeMessage();
+    } else {
+      super.postMessage(message, secondArg);
+    }
+  }
+
+  setPort2(port2: MessagePort): void {
+    this.port2 = port2;
+    this.postHandshakeMessage();
+  }
+
+  protected postHandshakeMessage() {
+    if (this.handshakeMessage && this.port2) {
+      super.postMessage(this.handshakeMessage, [this.port2]);
+    }
+  }
+}
+
+export class IFrameThreeDOMExecutionContext extends ThreeDOMExecutionContext {
+  constructor(capabilities: Array<ThreeDOMCapability>) {
+    super(capabilities);
+    const onMessageReceived = (event: MessageEvent) => {
+      switch (event.data.action) {
+        case 'handshakeRequest': {
```

Comment:
especially if used in multiple places, these string literals should be constants